src/repositories/khbooks/payment-repository.ts

```diff
@@ -39,3 +39,107 @@ export const countAllByFilters = async (where: Prisma.paymentsWhereInput) => {
   })
   return count
 }
+
+export const create = async (data: Prisma.paymentsUncheckedCreateInput) => {
+  return await prisma.payments.create({
+    data,
+  })
+}
+
+export const generatePaymentNumberById = async (id: number) => {
+  return await prisma.$transaction(async (tx) => {
+    const payments = await tx.payments.findMany({
+      where: {
+        AND: [
+          {
+            payment_no: {
+              not: undefined,
+            },
+          },
+          {
+            payment_no: {
+              not: null,
+            },
+          },
+          {
+            payment_no: {
+              not: "",
+            },
+          },
+        ],
+      },
+      orderBy: {
+        payment_no: "desc",
+      },
+      take: 1,
+    })
+
+    let paymentNo = Number(process.env.INVOICE_NO_OFFSET ?? 0)
```

Comment:
Payment number should only be 4 digits. Please create a separate env variable for this ^^

Reply:
I think no need for offset, since it will start at `0000` according to the ticket